### ISSUE TITLE
duplicated tag when using pytorch training U-Net on Images input and label.

### ISSUE THREAD

**Issue by Haonan-DONG** [FederatedAI/FATE#2960]
**Describe the bug**
We developed the pytorch code to training our Own model. And it came up with error :

**task_executor.py[line:191]: remote to [Party(role=arbiter, party_id=9996)] with duplicate tag: hash.b909faaadc5a0dd2ecbb.client_model.fit.default.model.0.**

We do not understand what the tag is.

**To Reproduce**
1. develop codes in the federateml/nn/backend/pytorch/data.py to fit image input and label.
2. develop codes in the federateml/nn/homo_nn/_torch.py to generate our own model.
3. run the data to train. And it came up with these error when eliminating the code for validation.

**Expected behavior**
Clearly, we successfully run through the demo with keras on csv data. And we also expect that the pytorch training will go as keras does. You know, training, and validating and testing.

**Screenshots**
![image](https://user-images.githubusercontent.com/42935157/128874490-c0475478-b7c1-4210-95ce-370662b2c093.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser chrome
 - Version 1.6

** Other Supplementory
We find that the official mnist demo with pytorch training will not work and it might stack at data alignment. Can it really run or not. 
Thanks for the excellent work!

**Comment by weiwee** [comment]
> **Describe the bug**
> We developed the pytorch code to training our Own model. And it came up with error :
> 
> **task_executor.py[line:191]: remote to [Party(role=arbiter, party_id=9996)] with duplicate tag: hash.b909faaadc5a0dd2ecbb.client_model.fit.default.model.0.**
> 
> We do not understand what the tag is.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
>     1. develop codes in the federateml/nn/backend/pytorch/data.py to fit image input and label.
> 
>     2. develop codes in the federateml/nn/homo_nn/_torch.py to generate our own model.
> 
>     3. run the data to train. And it came up with these error when eliminating the code for validation.
> 
> 
> **Expected behavior**
> Clearly, we successfully run through the demo with keras on csv data. And we also expect that the pytorch training will go as keras does. You know, training, and validating and testing.
> 
> **Screenshots**
> ![image](https://user-images.githubusercontent.com/42935157/128874490-c0475478-b7c1-4210-95ce-370662b2c093.png)
> 
> **Desktop (please complete the following information):**
> 
>     * OS: Ubuntu 18.04
> 
>     * Browser chrome
> 
>     * Version 1.6
> 
> 
> ** Other Supplementory
> We find that the official mnist demo with pytorch training will not work and it might stack at data alignment. Can it really run or not.
> Thanks for the excellent work!

remote/get resource is unique for transfervariable name + tag, use different tag to remote/get things if you remote/get things with same transfervariable

**Comment by Haonan-DONG** [comment]
Does "using the different tag to remote/get things" mean that we should get some change in our *_conf.json and *_dsl.json?

**Comment by Haonan-DONG** [comment]
> > **Describe the bug**
> > We developed the pytorch code to training our Own model. And it came up with error :
> > **task_executor.py[line:191]: remote to [Party(role=arbiter, party_id=9996)] with duplicate tag: hash.b909faaadc5a0dd2ecbb.client_model.fit.default.model.0.**
> > We do not understand what the tag is.
> > **To Reproduce**
> > Steps to reproduce the behavior:
> > ```
> > 1. develop codes in the federateml/nn/backend/pytorch/data.py to fit image input and label.
> > 
> > 2. develop codes in the federateml/nn/homo_nn/_torch.py to generate our own model.
> > 
> > 3. run the data to train. And it came up with these error when eliminating the code for validation.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **Expected behavior**
> > Clearly, we successfully run through the demo with keras on csv data. And we also expect that the pytorch training will go as keras does. You know, training, and validating and testing.
> > **Screenshots**
> > ![image](https://user-images.githubusercontent.com/42935157/128874490-c0475478-b7c1-4210-95ce-370662b2c093.png)
> > **Desktop (please complete the following information):**
> > ```
> > * OS: Ubuntu 18.04
> > 
> > * Browser chrome
> > 
> > * Version 1.6
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ** Other Supplementory
> > We find that the official mnist demo with pytorch training will not work and it might stack at data alignment. Can it really run or not.
> > Thanks for the excellent work!
> 
> remote/get resource is unique for transfervariable name + tag, use different tag to remote/get things if you remote/get things with same transfervariable

Does "using the different tag to remote/get things" mean that we should get some change in our *_conf.json and *_dsl.json?
